src/pages/projectSettings/tabs/EventLogTab/EventLogDiffs.ts

```diff
@@ -18,23 +18,23 @@ const getDiffProperties = (eventObj: object) => {
         ? event.concat(paths(value, fullPath))
         : event.concat(fullPath);
     }, []);
-
   return paths(eventObj);
 };
 
-const formatArrayElements = (eventKey: string) =>
+const formatArrayElements = (eventKey: string): string =>
   eventKey.replace(/.[0-9]./g, (x) => `[${x[1]}].`);
 
-const getNestedObject = (nestedObj: object, pathArr: string[]) =>
+const getNestedObject = (nestedObj: object, pathArr: string[]): EventValue =>
   pathArr.reduce(
     (obj, key) => (obj && obj[key] !== "undefined" ? obj[key] : undefined),
```

Comment:
This function confuses me. Like i get what it does but are there objects with the string "undefined" as the value instead of undefined?

Reply:
that check shouldn't be there, I removed it.